securedrop/source_app/main.py

```diff
@@ -51,34 +57,40 @@ def select_logo():
 
     @view.route('/create', methods=['POST'])
     def create():
-        filesystem_id = current_app.crypto_util.hash_codename(
-            session['codename'])
-
-        source = Source(filesystem_id, current_app.crypto_util.display_id())
-        db.session.add(source)
-        try:
-            db.session.commit()
-        except IntegrityError as e:
-            db.session.rollback()
-            current_app.logger.error(
-                "Attempt to create a source with duplicate codename: %s" %
-                (e,))
-
-            # Issue 2386: don't log in on duplicates
-            del session['codename']
-
-            # Issue 4361: Delete 'logged_in' if it's in the session
+        if logged_in():
+            flash(gettext("You have already logged-in from a different browser tab.\n"
+                          "Please verify your codename below as it may differ from "
+                          "the one displayed on the previous page"),
```

Comment:
nit: End sentence with '.'

Reply:
Good catch! I've added the period via 29f52d153194dceedd084dd95210c02ca276f5d2.